elastic/apm-server
# systemtest/cmd/sendotlp: add http/protobuf support

## Motivation/summary

Update `sendotlp` with a `-protocol` flag, which can be used to specify the transport protocol for sending OTLP data to APM Server. The flag value may be either "grpc" or "http/protobuf".

For now the protocol defaults to "grpc", but in the future (when OTLP/HTTP has been available in APM Server for a while) we should consider changing it to "http/protobuf" to align with the OpenTelemetry SDK default.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Start an APM Server 8.3.0+ deployment
2. Run `sendotlp -endpoint <apm-server-url> -secret-token <secret-token>`
3. Run `sendotlp -endpoint <apm-server-url> -secret-token <secret-token> -protocol http/protobuf`

Check that events show up both times, and check (e.g. through proxy logs) that the appropriate protocol is used.

## Related issues

None
